app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,29 @@
+module Katello
+  class RepositoryTypeManager
+    @repository_types = {}
+    class << self
+      attr_reader :repository_types
+      private :new
+      # Plugin constructor
+      def register(id, &block)
+        repository_type = ::Katello::RepositoryType.new(id)
```

Comment:
Should this check that the type being registered doesn't already exist in the `repository_types` object?